blog/2024-01-08-a-look-back-at-2023/index.mdx

```diff
@@ -0,0 +1,135 @@
+
+### Generating React prototypes based on your machines
+
+We also released the ability to [generate React app code](/docs/generate-react) based on your machines, which is a great way to prototype your app logic quickly. Stately AI adds more magic to your React prototypes with the option to **Enhance app** once generated, which builds out a UI for your app and enables you to go from a text description to a working prototype in less than a minute.
+
+<p>
+  <ThemedImage
+    alt="Enhancing an app in Stately’s editor, showing a pet grooming appointment app in the appointment booked state, with colored buttons generated for reschedule, cancel, and send reminder."
+    sources={{
+      light: '/blog/2024-01-08-a-look-back-at-2023/generate-react-app.png',
+      dark: '/blog/2024-01-08-a-look-back-at-2023/generate-react-app-dm.png',
+    }}
+  />
+</p>
+
+
+### Import any app logic
+
+We want to make visualizing app logic from any code as a state machine easy. So we’ve started with Zustand, Redux, and Redux Toolkit. You can now import your logic from any of these libraries into Stately Studio, and we’ll visualize it as a state machine. Import your existing app logic to generate up-to-date diagrams and customizable documentation for your logic, easily detect logical errors, and quickly generate test cases.
+
+### Adding assets to states
+
+Stately Studio has been able to add images to state and event descriptions for a while now, but you had to upload the images somewhere in the cloud and then add the image using markdown. Now, you can [drag an image on to a state](/docs/assets), and it’s automatically uploaded and added to your machine. And you’re not just limited to images; you can drag any file onto your state, and it will be displayed as a linked asset. Assets are a game-changer for improving documentation and easily linking to other resources.
+
+<p>
+  <ThemedImage
+    alt="State machine for a color picker where an image of an expanded color picker is being dragged into the open state."
+    sources={{
+      light: '/blog/2023-11-20-stately-studio-2-0/drag-assets.png',
+      dark: '/blog/2023-11-20-stately-studio-2-0/drag-assets-dm.png',
+    }}
+  />
+</p>
+
+### Stately inspector
+
+Since Stately Studio succeeded the old Stately viz, we’ve been working to bring back some of our favorite features from the viz into the editor. One of the last things we released in 2023 was a new package for [@statelyai/inspect](https://github.com/statelyai/inspect), which builds upon the old XState inspector, enabling developers to visually inspect state transitions *and* actor communication in any JavaScript/TypeScript app. You can visualize your state machines and generate sequence diagrams for how your actors communicate. You can use Stately Inspect with or without XState, with any framework or library, or none at all. [Watch our last office hours from 2023 for David’s demo of the Stately inspector](https://www.youtube.com/watch?v=x-F1TPjz_lI&t=1232s).
```

Comment:
```suggestion
Since Stately Studio succeeded the old Stately viz, we’ve been working to bring back some of our favorite features from the viz into the editor. One of the last things we released in 2023 was a new package for [@statelyai/inspect](https://github.com/statelyai/inspect), which builds upon the old XState inspector, enabling developers to visually inspect state transitions *and* actor communication in any JavaScript/TypeScript app. You can visualize your state machines and generate sequence diagrams for how your actors communicate. You can use Stately Inspect with or without XState, with any framework or library, or none at all. [Watch our last office hours from 2023 for David’s demo of the Stately inspector](https://www.youtube.com/watch?v=x-F1TPjz_lI&t=103s).
```
Stately inpect demo starts at 103s